### ISSUE TITLE
Custom remote entry file name not getting set in mf-manifest.json

### ISSUE THREAD

**Issue by DeveshSapkale** [module-federation/vite#226]
@gioboa 
Issue:
When I use below configuration in federation plugin using vite remote from the example given in this repo
{ 
...
manifest: true
fileName: test.remoteEntry.js
....
 } 

The mf-manifest.json gets generated after vite build, but in
Manifest.json
{
.....
"remoteEntry ": {
"path": "",
"type: "module"
}
...
}

File name details in this manifest are empty for remoteEntry object. 

This detail is only available when I keep name as remoteEntry.js

I debugged few line of code in the plugin repo and found that in file named "src/plugins/pluginMFManifest.ts"
Under generate bundle() function there is logic which sets "remoteEntryFile" variable which is then used to set the file name in mf-manifest.json file.

In this code we check if mfOptions.filename == fileData.name which is never true because mfOptions.filename will be a full name of "test.remoteEntry.js" with .js ext.
But fileData.name will be without js ext which is just "test.remoteEntry"

So from fix point of view I think condition should use fileData.preliminaryName instead of fileData.name.

But I am not aware of the usages and reason behind that code.

Are we aware of this issue or is there a specific way I need to provide the filename in federation plugin configuration.

I need to use custom name for it, that's a requirement.

**Comment by gioboa** [comment]
Hi @DeveshSapkale thanks for your feedback and investigation.
May I ask you to tackle this issue? Thanks

**Comment by DeveshSapkale** [comment]
@gioboa Do you mean I should create a PR for this fix. Or should handle it at my end which means simply using "remoteEntry.js" name?
Thanks for the quick resonse

**Comment by gioboa** [comment]
I really would like to give you the possibility to cover your needs. Is it possible with we pack/rspack? Because in the end we are using the core library under the hood

**Comment by DeveshSapkale** [comment]
Having custom name for remoteEntry file is possible with rspack's module federation plugin, it creates mf-manifest.json file properly, with given custom remote entry filename. 
I agree that under the hood core library is getting used, but creation of mf-manifest.json file is part of this repo.

**Comment by gioboa** [comment]
Ok so let's try to implement it

**Comment by zhangHongEn** [comment]
try `test-remoteEntry.js`

**Comment by DeveshSapkale** [comment]
@gioboa @zhangHongEn I agree that "test-remoteEntry.js" might work but I am using "test.remoteEntry.js" for my application as a convention "{remote name}.remoteEntry.js".
Sorry for the late reply I will try to create a PR soon please review it , as I am not fully aware of of this code base.

**Comment by gioboa** [comment]
I think the double dot can mess up things, that's my guess
